iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/VoltageLevelImpl.java

```diff
@@ -618,4 +619,91 @@ public void allocateVariantArrayElement(int[] indexes, int sourceIndex) {
         topologyModel.allocateVariantArrayElement(indexes, sourceIndex);
     }
+
+    @Override
+    public void convertToTopology(TopologyKind newTopologyKind) {
+        Objects.requireNonNull(newTopologyKind);
+        if (newTopologyKind != topologyModel.getTopologyKind()) {
+            if (newTopologyKind == TopologyKind.NODE_BREAKER) {
+                throw new PowsyblException("Topology model conversion from bus/breaker to node/breaker not yet supported");
+            } else if (newTopologyKind == TopologyKind.BUS_BREAKER) {
+                BusBreakerTopologyModel newTopologyModel = new BusBreakerTopologyModel(this);
+
+                for (Bus bus : topologyModel.getBusBreakerView().getBuses()) {
+                    // no notification, this is just a mutation of a calculation bus to a configured bus
+                    ConfiguredBusImpl configuredBus = new ConfiguredBusImpl(bus.getId(), bus.getOptionalName().orElse(null), bus.isFictitious(), this);
+                    newTopologyModel.addBus(configuredBus);
+                }
+
+                // transfer retained switches
+                for (Switch sw : topologyModel.getBusBreakerView().getSwitchStream().toList()) {
+                    String busId1 = topologyModel.getBusBreakerView().getBus1(sw.getId()).getId();
+                    String busId2 = topologyModel.getBusBreakerView().getBus2(sw.getId()).getId();
+                    // no notification, this is just a transfer
+                    ((NodeBreakerTopologyModel) topologyModel).removeSwitchFromTopology(sw.getId(), false);
+                    newTopologyModel.addSwitchToTopology((SwitchImpl) sw, busId1, busId2);
+                }
+
+                // reconnect all connectable to new topology model
+                // first store all bus/breaker topological infos associated to this terminal because we will start moving
+                // terminal from old mode to new one, it will modify the old topology model
+                List<Triple<TerminalExt, Bus, Boolean>> oldTopologyModelInfos = new ArrayList<>();
+                for (Terminal oldTerminal : topologyModel.getTerminals()) {
+                    Bus connectableBus = oldTerminal.getBusBreakerView().getConnectableBus();
+                    boolean connected = oldTerminal.isConnected();
+                    oldTopologyModelInfos.add(Triple.of((TerminalExt) oldTerminal, connectableBus, connected));
+                }
+
+                for (var triple : oldTopologyModelInfos) {
+                    TerminalExt oldTerminalExt = triple.getLeft();
+                    Bus connectableBus = triple.getMiddle();
+                    boolean connected = triple.getRight();
+
+                    // if there is no way to find a connectable bus, remove the connectable
+                    // an alternative would be to connect them all to a new trash configured bus
+                    if (connectableBus != null) {
```

Comment:
Isn't there's always a non-null connectableBus? A connectable is tied to a node, and a CalculatedBus is always created for a non-empty set of nodes:
https://github.com/powsybl/powsybl-core/blob/218eb5055c456a913c28199f435a0b49e468ce92/iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerTopologyModel.java#L491-L496

Reply:
I'm almost sure I already seen null connectable bus ID in the bus/breaker view of a node/breaker topology when all switches are open.